src/Analysis/Ast/Impl/Documents/DocumentBuffer.cs

```diff
@@ -78,20 +72,33 @@ internal sealed class DocumentBuffer {
             }
         }
 
-        private static IEnumerable<NewLineLocation> SplitLines(StringBuilder text) {
-            NewLineLocation nextLine;
-
-            // TODO: Avoid string allocation by operating directly on StringBuilder
-            var str = text.ToString();
+        public IEnumerable<NewLineLocation> SplitLines() {
```

Comment:
Does this need to be public?

Reply:
The class is internal so it is internal - for the tests access. Perhaps rename it to `GetNewLineLocations` instead?

Reply:
Ah, didn't realize the type was internal. Whatever you want.

Reply:
Renamed